snap/snapcraft.yaml

```diff
@@ -70,71 +74,57 @@ apps:
 grade: stable
 confinement: classic
 
-parts:
-  # The cryptography library (indirect dependency via craft-store) requires rust, but only ships wheels for amd64 and arm64.
-  # As of cryptography 39, it requires rust v1.48.0 or newer, which is available from focal-updates.
-  # This part builds the wheel for cryptography.
-  cryptography-deps:
-    plugin: nil
-    source: .
-    build-packages:
-      - cargo
-      - rustc
-      - pkg-config
-      - python3.8-dev
-      - libpython3.8-dev
-      - libssl-dev
-      - libffi-dev
-      - python3-wheel
-      - python3-pip
-      - python3-setuptools-scm
-    override-build: |
-      # Dirty workaround - builds the cryptography wheel for later installation.
-      # This must be exactly the version referenced in requirements.txt, so this is fragile.
-      pip wheel $(grep '^cryptography[<=>]' requirements.txt)
+build-packages:
+  - cargo
+  - rustc
+  - pkg-config
+  - python3.10-dev
+  - libpython3.10-dev
+  - libssl-dev
+  - libffi-dev
+  - python3-wheel
+  - python3-pip
+  - python3-setuptools-scm
+
 
-  # Classic core20 snaps require staged python.
+parts:
+  # Classic core22 snaps require staged python.
   python3:
-    after: [cryptography-deps]
     plugin: nil
+    build-attributes:
+      - enable-patchelf
     build-packages:
       - python3-dev
     stage-packages:
       - libpython3-stdlib
-      - libpython3.8-minimal
-      - libpython3.8-stdlib
-      - python3.8-minimal
+      - libpython3.10-minimal
+      - libpython3.10-stdlib
+      - python3.10-minimal
       - python3-distutils
       - python3-minimal
       - python3-pip
+      - python3-pip-whl
       - python3-setuptools
+      - python3-setuptools-whl
       - python3-venv
       - python3-wheel
     override-build: |
-      snapcraftctl build
-      install -D -m 0755 $SNAPCRAFT_PROJECT_DIR/snap/local/sitecustomize.py $SNAPCRAFT_PART_INSTALL/usr/lib/python3.8/sitecustomize.py
-
-  python-apt:
-    after: [python3]
-    plugin: nil
-    source: .
-    build-packages:
-      - libapt-pkg-dev
-      - python3.8-dev
-      - libpython3.8-dev
-    build-environment:
-      - LDFLAGS: -L/usr/lib/python3.8
-      - CPPFLAGS: -I/usr/include/python3.8
-    override-build: |
-      pip wheel $(cat requirements-focal.txt)
+      ${SNAP}/libexec/snapcraft/craftctl default
+      install -D -m 0755 $CRAFT_PROJECT_DIR/snap/local/sitecustomize.py $CRAFT_PART_INSTALL/usr/lib/python3.10/sitecustomize.py
+    override-prime: |
+      ${SNAP}/libexec/snapcraft/craftctl default
+      # override python default venv wheels dir
+      sed -i "s/sysconfig.get_config_var('WHEEL_PKG_DIR')/'\/snap\/charmcraft\/current\/usr\/share\/python-wheels\/'/" ${CRAFT_PRIME}/usr/lib/python3.10/ensurepip/__init__.py
```

Comment:
why do we need this?

Reply:
We no longer need this as it fixed in the new version